src/sentry/notifications/helpers.py

```diff
@@ -833,9 +833,141 @@ def get_setting_options_for_recipient(
     return notification_settings
 
 
+def get_setting_options_for_users(
+    user_ids: Iterable[int],
+    project: Project | None = None,
+    organization: Organization | None = None,
+) -> MutableMapping[
+    RpcActor, MutableMapping[NotificationSettingEnum, NotificationSettingsOptionEnum]
+]:
+    """
+    Returns a map of users to NotificationSettingOption by type:
+    {user -> {notification type -> setting}}.
+
+    Args:
+        user_ids: The user ids to get notification settings for.
+        project: The project to get notification settings for.
+        organization: The organization to get notification settings for.
+    """
+    query = get_query(project=project, organization=organization, user_ids=user_ids)
+    notification_settings = NotificationSettingOption.objects.filter(query)
+
+    user_to_setting: MutableMapping[
+        RpcActor, MutableMapping[NotificationSettingEnum, NotificationSettingsOptionEnum]
+    ] = defaultdict(dict)
+    for ns in notification_settings:
```

Comment:
@snigdhas nit: maybe this would be easier to read if we just had one for loop and use if/elif